src/status_im/common/enter_seed_phrase/view.cljs

```diff
@@ -51,43 +51,36 @@
       (string/replace #"\s+" " ")
       (string/trim)))
 
-(defn- recovery-form
-  [{:keys [seed-phrase word-count error-state? all-words-valid? on-change-seed-phrase
-           keyboard-shown? on-submit]}]
-  (let [button-disabled? (or error-state?
-                             (not (constants/seed-phrase-valid-length word-count))
-                             (not all-words-valid?))]
-    [rn/view {:style style/form-container}
-     [header word-count]
-     [rn/view {:style style/input-container}
-      [quo/recovery-phrase-input
-       {:accessibility-label      :passphrase-input
-        :placeholder              (i18n/label :t/seed-phrase-placeholder)
-        :placeholder-text-color   colors/white-opa-30
-        :auto-capitalize          :none
-        :auto-correct             false
-        :auto-focus               true
-        :mark-errors?             true
-        :word-limit               max-seed-phrase-length
-        :error-pred-current-word  partial-word-not-in-dictionary?
-        :error-pred-written-words word-not-in-dictionary?
-        :on-change-text           on-change-seed-phrase}
-       seed-phrase]]
-     [quo/button
-      {:container-style (style/continue-button keyboard-shown?)
-       :type            :primary
-       :disabled?       button-disabled?
-       :on-press        on-submit}
-      (i18n/label :t/continue)]]))
+(defn- recovery-phrase-form
+  [{:keys [title seed-phrase word-count on-change-seed-phrase on-input-ref]} & children]
+  (->> children
+       (into
+        [rn/view {:style style/form-container}
+         [header title word-count]
+         [rn/view {:style style/input-container}
+          [quo/recovery-phrase-input
+           {:accessibility-label      :passphrase-input
+            :on-input-ref             on-input-ref
+            :placeholder              (i18n/label :t/seed-phrase-placeholder)
+            :placeholder-text-color   colors/white-opa-30
+            :auto-capitalize          :none
+            :auto-correct             false
+            :auto-focus               true
+            :mark-errors?             true
+            :word-limit               max-seed-phrase-length
+            :error-pred-current-word  partial-word-not-in-dictionary?
```

Comment:
Here the recovery-phrase-form component has been refactored to receive `children` as an argument. This allows us to customise the submit buttons for the form without needing to pass more props into this component.